### PR TITLE
Allow Sort as type annotation

### PR DESCRIPTION
**Pull request by sjakobi**
Standardized in https://github.com/dhall-lang/dhall-lang/pull/507.

### CODE REVIEW

File: dhall/src/Dhall/TypeCheck.hs [hunk 1]
```diff
@@ -195,7 +196,9 @@ typeWithA tpa = loop
             loop ctx b2
 
     loop ctx e@(Annot x t       ) = do
-        _ <- loop ctx t
+        case Dhall.Core.denote t of
+            Const Sort -> return ()
```
Comment (sjakobi) [inline dhall/src/Dhall/TypeCheck.hs]: Would that be a good idea?

```suggestion
            Const _ -> return ()
```

Comment (Gabriel439) [reply dhall/src/Dhall/TypeCheck.hs]: Yeah, there's no reason to type-check `t` further if it is already a `Const`

**Comment by sjakobi** [comment]
Thanks for the commit bits @Gabriel439! :)

Given your approval, can I just merge once CI has passed?

**Comment by Gabriel439** [comment]
@sjakobi: Yeah, you can merge at any time once you have an approval

**Pull request merged**